Gifski/Utilities.swift

```diff
@@ -3684,3 +3672,103 @@ extension CGImage {
 	static let empty = NSImage(size: CGSize(widthHeight: 1), flipped: false) { _ in true }
 		.cgImage(forProposedRect: nil, context: nil, hints: nil)!
 }
+
+
+extension CGImage {
+	var size: CGSize { CGSize(width: width, height: height) }
+
+	var hasAlphaChannel: Bool {
+		switch alphaInfo {
+		case .first, .last, .premultipliedFirst, .premultipliedLast:
+			return true
+		default:
+			return false
+		}
+	}
+}
+
+
+extension CGImage {
+	/**
+	Returns a read-only pointer to the bytes of the image.
+
+	- Important: Don't assume the format of the underlaying storage. It could be `ARGB`, but it could also be `RGBA`. Draw the image into a `CGContext` first to be safe. See `CGImage#converting`.
+	*/
+	var bytePointer: UnsafePointer<UInt8>? {
+		guard let data = dataProvider?.data else {
+			return nil
+		}
+
+		return CFDataGetBytePtr(data)
+	}
+}
+
+
+extension CGContext {
+	/// Create an `ARGB` bitmap context.
+	static func argbBitmapContext(width: Int, height: Int, withAlpha: Bool) -> CGContext? {
+		let alphaInfo = withAlpha ? CGImageAlphaInfo.premultipliedFirst : .noneSkipFirst
```

Comment:
Premultiplied color space is not compatible with regular uncorrelated RGBA/ARGB and will darken semitransparent colors.

You will need to divide RGB components by A to undo this.

Reply:
Are you saying that libgifski accepts non-premultiplied ARGB? If so, I could add a `premuliplied` option and use https://developer.apple.com/documentation/coregraphics/cgimagealphainfo/first for that. Correct?

Reply:
Yes, gifski accepts only non-premultiplied colors. Internally it works with uncorrelated RGBA *only*. 

gifski's ARGB function call also takes uncorrelated ARGB, and just swaps bytes to call its own RGBA function.

I think `.first` would be technically correct. But beware - last time I tried, macOS didn't support drawing to non-premultiplied bitmaps.

Reply:
> gifski accepts only non-premultiplied colors

Maybe you should make that explicit in the libgifski docs?

> I think .first would be technically correct. But beware - last time I tried, macOS didn't support drawing to non-premultiplied bitmaps.

Yeah, I just discovered that myself... I guess I'll have to do it manually with a loop.

---

So Gifski.app has been wrong this whole time?

Reply:
I haven't checked, but yeah, very likely that it was incorrect.

Reply:
> I haven't checked, but yeah, very likely that it was incorrectly premultiplied.

I can confirm that this was indeed a problem. I just tried with a video with transparency and the colors looked much darker on the current Gifski version.